src/template-tags/tickets.php

```diff
@@ -174,119 +176,63 @@ function tribe_tickets_display_count_and_ticket_button() {
 		$cta_html = '';
 
 		// If we have tickets or RSVP, but everything is Sold Out then display the Sold Out message
-		if ( ( $types['tickets']['count'] || $types['rsvp']['count'] ) && ( ! $types['tickets']['available'] && ! $types['rsvp']['available'] ) ) {
-
-			$cta_html = '<span class="tribe-out-of-stock">' . esc_html_x( 'Sold out', 'list view stock sold out', 'event-tickets' ) . '</span>';
-		} else {
-
-			$rsvp = false;
-			// Determine button text and anchor link based on an events tickets
-			if (
-				$types['tickets']['count']
-				&& $types['tickets']['available']
-				&& 0 === $types['rsvp']['count']
-			) {
-				// if tickets and stock with no rsvp
-				$rsvp = false;
-			} elseif (
-				$types['tickets']['count']
-				&& $types['tickets']['available']
-				&& $types['rsvp']['count']
-				&& 0 <= $types['rsvp']['available']
-			) {
-				// if tickets and rsvp with stock in both or rsvp sold out
-				$rsvp = false;
-			} elseif (
-				0 === $types['tickets']['count']
-				&& $types['rsvp']['count']
-				&& $types['rsvp']['available']
-			) {
-				// if no tickets and rsvp available
-				$rsvp = true;
-			} elseif (
-				$types['tickets']['count']
-				&& 0 <= $types['tickets']['available']
-				&& $types['rsvp']['count']
-				&& $types['rsvp']['available']
-			) {
-				// if no tickets and rsvp available
-				$rsvp = true;
+		foreach ( $types as $type => $data ) {
+			if ( ! $data['count'] ) {
+				continue;
 			}
 
-			$stock = $types['tickets']['stock'];
-			if ( $rsvp ) {
-				$stock = $types['rsvp']['stock'];
-			}
+			if ( $data['count'] && ! $data['available'] ) {
+				$cta_html .= '<span class="tribe-out-of-stock">' . esc_html_x( 'Sold out', 'list view stock sold out', 'event-tickets' ) . '</span>';
+			} else {
+				$stock = $data['stock'];
+				if (
+					$data['unlimited'] ||
```

Comment:
`||` should move to the next line (reasons [given here](http://moderntribe.github.io/products-engineering/php/#php-control-multi))